Add data management section (#84)
sa-tre/satre-specification: docs/source/standard.md

The ability of the TRE organisation to manage how and where data is stored, how it moves, changes and is removed.

```{list-table}
:header-rows: 1
:name: tab-data-lifecycle-management

* - Statement
  - Guidance

> * - A TRE must have a data ingress process which enforces information governance rules/processess.

Reviewer: ```suggestion
* - A TRE must have a data ingress process which enforces information governance rules/processes.
```

  - The data ingress process needs to ensure that information governance is correctly followed.

>     in particular, it should require that an ingress request has been approved by all required par ties.

Reviewer 2: ```suggestion
    in particular, it should require that an ingress request has been approved by all required parties.
```

> * - A TRE must have a data egress process which enforces information governance rules/processess.

Reviewer: ```suggestion
* - A TRE must have a data egress process which enforces information governance rules/processes.
```

>   - The data egress process needs to ensure that information governance is correctly followed.
>     in particular, it should require that an egress request has been approved by all required par ties.

Reviewer: ```suggestion
  - The data egress process needs to ensure that information governance requirements are adhered to.
    In particular, it should require that an egress request has been approved by all required parties.
```

* - A TRE could require independent approval for data ingress and egress events.

>   - A person indepdent to a project, for example a TRE admin or dedicated referee, could check ingress and egress requests to ensure information governance is correctly followed in the absence of an automated process.

Reviewer: ```suggestion
  - A person independent to a project, for example a TRE admin or dedicated referee, could check ingress and egress requests to ensure information governance processes are correctly followed in the absence of an automated process.
```

>   - A person independent to a project, for example a TRE admin or dedicated referee, could check ingress and egress requests to ensure information governance processes are correctly followed in the absence of an automated process.

Reviewer 2: > in the absence of an automated process

This reads like there's an expectation of an automated process, does any TRE currently have one?

Also (for the future)- we need to agree what "project" means. E.g. is the data provider part of the project? If so they should be allowed to authorise egress, even though they're not independent from the project.

Reviewer: Our data handling procedures assume that:
- the TRE hosting organisation (Turing in our case)
- the project team (may or may not be Turing employees)
- each of the data providers
are stakeholders in the data analysis and should have a say in what gets ingressed and egressed.

Author: So, for now, are we all OK with this if

1. Reference to an automated process is removed
2. We leave 'project' to be defined or adjusted later
3. Add some context to _why_ the TRE organisation may want to (or need to) have input on egress approval?

Reviewer 2: Yes, I think focussing on the requirement (there should be _a_ process) and the main reasons/context for _why_ a process is required makes sense.

* - A TRE must keep a record of what data it holds.
  - Good records are important for ensuring compliance with legislation, understanding risk and aiding good data hygiene.

>     The record should include a description of the data, its source, contact details for the data owner, which projects use the data, the data it was recieved, when it is expected to no longer be needed.

Reviewer 2: ```suggestion
    The record should include a description of the data, its source, contact details for the data owner, which projects use the data, the date it was recieved, when it is expected to no longer be needed.
```
May need to update this in future to include streaming data, or datasets that are regularly updated.

* - A TRE must have a policy on data deletion.

>   - There should be a clear, published policy on when data will be deleted beyond at the conclusion of a project.

Reviewer 2: ```suggestion
  - There should be a clear, published policy on when data will be retained or deleted.
```
Data sources may be separate from the projects using them.

    This may allow time for data owners to consider outputs they may want to extract from the TRE.
    Any sensitive data, including all backups, should be deleted when they are no longer needed.

>     Having clear policy will help to avoid problems with data being kept longer than necessary or accidental deletion of outputs.

Reviewer 2: ```suggestion
    Having clear policies will help to avoid problems with data being kept longer than necessary or accidental deletion of outputs.
```
or `a clear policy`?

* - A TRE could keep backups of data and research environments
  - Keeping backups could help reduce the impact of events like accidental deletion and data corruption on work in a TRE.
    TRE developers may want to consider how different elements, for example sensitive input data or users workspaces, may be backed up or if they should be.
* - A TRE should log how input data is modified.
  - If the input data is mutable a TRE should keep records of its modification.

>     For example when the data was modified and by who.

Reviewer: ```suggestion
    For example, when the data was modified and by whom.
```

Author: I did think about using whom. This will be a fairly formal document.

Normally I choose to never use in writing whom because it is rarely used now and sounds a bit stuffy.

Author: resolved in [1d85fa7](https://github.com/sa-tre/satre-specification/pull/84/commits/1d85fa7ee4574d6963763f0fe6c9ead7b15c4bba)

* - A TRE must, to a reasonable extent, prevent unathorised data ingress or egress
  - Movement of data which has not been subject to information governance processes risks breaking rules and is more likely to result in a data breach.

>     However, it is difficult to control for every possibility.
>     For example, a user may take pictures of their computer screen to remove data, or using a device presnting as a USB HID keyboard to input large amounts of text.

Reviewer: ```suggestion
    However, it is difficult to control for every possibility, for example, a user may take pictures of their computer screen to remove data, or using a device presenting as a USB HID keyboard to input large amounts of text.
```

Author: Is there a grammatical problem here? I prefer to avoid long sentences if possible.

I think I did mess up tenses here though "a user _may take_ pictures... or _using_ a device" should be "a user _may take_ pictures... or _use_ a device".

Reviewer: `However, it is difficult to control for every possibility.` felt like a sentence fragment that should be joined onto another sentence. Also `presnting -> presenting`.

Author: I'm trying to channel my PhD supervisor. I think it is a sentence (subject, verb) so I would prefer to avoid joining it if it isn't necessary. Not sure if this is a hill I will die on :smile:.

Author: Changes in [b6c00fe](https://github.com/sa-tre/satre-specification/pull/84/commits/b6c00fe61c61ac69c19fee09ab45528ce848b96a)

>     An example of a reasonable measure would be for a remote desktop based TRE to prevent data being shared between a workspace's and local machine's clipboards.

Reviewer 2: ```suggestion
    An example of a reasonable measure would be for a remote desktop based TRE to prevent data being copied from a local machine's clipboard to a workspace.
```
xref: https://github.com/sa-tre/satre-specification/pull/89/files#r1221347970

Author: Just to double check this.

The other section says **must not** allow copy/paste from inside a TRE to outside.
This section says **must** reasonably prevent unauthorised data movement, which **could** include blocking copy/paste from outside a TRE to inside.

```

### 3.2 Identity and access management

The ability of the TRE organisation to ensure the right people (identities) can access the tools and data they need and no more.

```{list-table}
:header-rows: 1
:name: tab-identity-and-access-management

* - Statement
  - Guidance
* - A TRE must not create user accounts for use by more than one person.
  - It is important that each user account should be used by one, and only one, person in order to facilitate the assignment of roles or permissions and to log the actions of individuals.
* - A TRE must be reasonably convinced of the identity of the person being granted an account.
  - It is important to ensure access, via an account, has been given to the correct person.

>     For example, multiple credentials may be used in account creation to verify identitity or, when appropriate, photo ID checks may be required.

Reviewer 2: ```suggestion
    For example, multiple credentials may be used before account creation to verify identity or, when appropriate, photo ID checks may be required.
```

* - A TRE must ensure multi-factor authentication for users.
  - Multi-factor authentication ensures that to successfully connect a user must have more than one piece of evidence in different categories.
    Categories include something the user knows (_e.g._ a password), something the user possesses (_e.g._ a TOTP key) or something the user is (_e.g._ biometric data).
    A TRE does not need to implement multi-factor authentication checks itself if it is provided by a third-party identity provider.
* - A TRE could restrict access to particular locations.
  - Restricting access to a set of known, static, personal or institutional IP addresses can help avoid speculative attacks.
    When appropriate, access could also be restricted to physical locations with security controls and access requirements.
```

### 3.3 Output management

The ability of the TRE organisation to ensure outputs are safely published and shared.

```{list-table}
:header-rows: 1
:name: tab-output-management

* - Statement
  - Guidance
* - A TRE should have a system to aid classifying outputs.
  - Removing data from a TRE can be a difficult process as there is potential for sensitive data to be revealed.

>     Having guidance, processess and methods to help will ensure that outputs are correctly classified and, furthermore, that outputs when may be openly published are identified.

Reviewer 2: Not sure about the wording here, I've tried to make a suggestion.

>     Encouraging openly published outputs will enhance a TREs impact over handing all outputs to the data provider.

Reviewer: ```suggestion
    Encouraging openly published outputs rather than handing all outputs to the data provider will enhance a TRE's impact .
```

> * - A TRE should establish projects intended outputs from the outset.

Reviewer: ```suggestion
* - A TRE should establish each project's intended outputs from the outset.
```

>   - Identifying the purpose of a piece of work is important for compliance with data protection legilation.

Reviewer 2: ```suggestion
  - Identifying the purpose of a piece of work is important for compliance with data protection legislation.
```

>     Results will be produced which address the projects purpose, some of which may be outputs removed from the TRE.

Reviewer 2: ```suggestion
    Results will be produced which address the project's purpose, some of which may be outputs that are removed from the TRE.
```
